vizro-core/tests/unit/vizro/models/_components/form/test_radioitems.py

```diff
@@ -0,0 +1,144 @@
+                value="A",
+                className="selector_body_radio_items",
+                persistence=True,
+            ),
+        ],
+        className="selector_container",
+    )
+
+
+class TestRadioItemsInstantiation:
+    """Tests model instantiation."""
+
+    def test_create_radio_items_mandatory_only(self):
+        radio_items = RadioItems()
+
+        assert hasattr(radio_items, "id")
+        assert radio_items.type == "radio_items"
+        assert radio_items.options == []
+        assert radio_items.value is None
+        assert radio_items.title is None
+        assert radio_items.actions == []
+
+    def test_create_radio_items_mandatory_and_optional(self):
+        radio_items = RadioItems(options=["A", "B", "C"], value="A", title="Title", id="radio_items_id")
+
+        assert radio_items.id == "radio_items_id"
+        assert radio_items.type == "radio_items"
+        assert radio_items.options == ["A", "B", "C"]
+        assert radio_items.value == "A"
+        assert radio_items.title == "Title"
+        assert radio_items.actions == []
+
+    @pytest.mark.parametrize(
+        "test_options, expected",
+        [
+            (["A", "B", "C"], ["A", "B", "C"]),
+            ([1, 2, 3], [1.0, 2.0, 3.0]),
+            ([1.0, 2.0, 3.0], [1.0, 2.0, 3.0]),
+            ([True, False, True], [True, False, True]),
+            (
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}],
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}],
+            ),
+            (
+                [{"label": "New York", "value": "NYC"}, {"label": "Berlin", "value": "BER"}],
+                [{"label": "New York", "value": "NYC"}, {"label": "Berlin", "value": "BER"}],
+            ),
+            (
+                [{"label": "True", "value": True}, {"label": "False", "value": False}],
+                [{"label": "True", "value": True}, {"label": "False", "value": False}],
+            ),
+            ([True, 2.0, 1.0, "A", "B"], ["True", "2.0", "1.0", "A", "B"]),
+        ],
+    )
+    def test_create_radio_items_valid_options(self, test_options, expected):
+        radio_items = RadioItems(options=test_options)
+
+        assert hasattr(radio_items, "id")
+        assert radio_items.type == "radio_items"
+        assert radio_items.options == expected
+        assert radio_items.value is None
+        assert radio_items.title is None
+        assert radio_items.actions == []
+
+    @pytest.mark.parametrize("options", [1, "A", True, 1.0])
+    def test_create_radio_items_invalid_options_type(self, options):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            RadioItems(options=options)
+
+    def test_create_radio_items_invalid_options_dict(self):
+        with pytest.raises(
+            ValidationError,
+            match="Invalid argument `options` passed into Selector. Expected a dict with keys `label` and `value`.",
```

Comment:
Once, we support a RadioItem to be configured on the right side of the screen (and not only as control selector), should we then change the word "Selector" to "component" or "model" inside the ValidationError message? 

If yes, should we do it now then?

Reply:
That's a good question, I don't think we would call it like that though as these terms are also too generic. Let me just remove the "into Selector" part and then it should always be valid.